### DISCUSSION TITLE
[GUIDE] How to create and publish "language packs"

### DISCUSSION THREAD

**Discussion by Lailloken** [Lailloken/Exile-UI#326] [Translations / Localization]
Community translations will not be officially implemented into the project because I personally can't really provide any valuable input for maintaining or testing them (i.e. I would probably just slow things down). Instead, they will be separate projects led by however many people want to contribute to making the tool usable in their language.

The required files will be mirror-hosted on a different [GitHub repository](https://github.com/Lailloken/Lailloken-UI-English)  in order to facilitate tracking their changes. These serve as a template to create new "language packs" for the desired target languages. On that repository, there will also be detailed instructions which should be helpful with starting such a project.

It's a good idea to advertise/announce your plans in a new thread in the [Translations / Localization](https://github.com/Lailloken/Lailloken-UI/discussions/categories/translations-localization) section in order to prevent multiple smaller projects from emerging, working towards the same goal independently. It would be more efficient if people grouped up and worked together, and it would also help me stay up-to-date (so I can make sure the required files are in good shape).

## Publication
I kindly ask you to not publish them as part of a fork of the main project. Simply host your language-files on your own GitHub repository.

**Comment by vagrant-soul** [comment]
I'd like to suggest poe2 as a new project in its own right, so that we can perhaps reduce the amount of work we have to do when localising the language.
I found that there are a lot of poe1 data files, if these files are not translated, will it affect the use of the poe2 part?
Sorry, my native language is not English, so I can't express myself very accurately.

**Reply by Lailloken** [reply]
I believe it's more efficient to keep them together because both games use the same "foundation", i.e. tooltips, log-files, game mechanics, and many more aspects work more or less in the same way. Many tool-features can be used in both games without major changes.
***
Translators can decide if they want to translate the UI, or make the tool compliant with non-English clients, or both. If UI aspects are missing from the translated files, it will fall back to the original English ones and display those instead. If client-related strings are missing, those feature that depend on them will simply not work.

If someone only wanted to translate/localize for PoE2, they could simply leave out PoE1-related stuff, and vice versa. The files are segmented into groups by features, and most keys should be tagged clearly enough to deduce which game uses them (e.g. ` maps_shaper `, ` maps_exarch `)
***
There's currently an issue with PoE2: there's no way to check which language the client is using. So I would have to implement a language-selection first.

I'm still not sure I want to continue supporting localizations because my tool is rather niche and predominantly played in English (at least in western parts). It's been over a year since I implemented support based on user-requests, and no-one has followed up on their initial plans. Even though it was somewhat interesting to implement this, I feel like I have been "baited" into this.

**Comment by vagrant-soul** [comment]
In poe2, this part of the settings in file poe2_production_Config contains the client's current language settings
`[LANGUAGE]
language=en`
This image is my language setting
![Snipaste_2025-01-12_00-28-12](https://github.com/user-attachments/assets/fe358875-fd8d-43fc-af94-8acea52cfb0a)

I tried to do a small part of the test, but I found that analyzing Client.txt is not as easy as I thought, and I didn't understand the basic functions of LL-ui before, and I had to switch between English and other languages very often, very frustrating!
![Snipaste_2025-01-11_23-34-12](https://github.com/user-attachments/assets/20962f86-60ba-4db6-8cf5-62b3cad55067)
I think I first need to learn the code and use the basic functions before trying the next step of localization
Thanks for the tools.

**Reply by Lailloken** [reply]
I see, the language-setting is only stored in the ini-file if you change it at least once, otherwise it's simply ` blank = English `.
EDIT: I tried changing the language during launch-evening, and it didn't work, so I didn't look into it further.

Your initial test looks decent. I don't think you need to know the code to proceed further, it's just that the files on the separate GitHub repo have not been maintained lately because no-one showed any interest in working on a translation.

Before you proceed, I should check everything again and maybe also tidy things up a bit for PoE2.
***
Your log-file health shows ` 100% (0 MB / 0 MS) `, which may indicate that the tool couldn't find your log-file. I know the Korean client uses a different log-file, does the same apply to you? Could you check if it's ` X:\Path of Exile 2\logs\Client.txt `?

**Comment by vagrant-soul** [comment]
The log file is fine, it shows 100% because I tried a different language switch and deleted the previous logs to make it easier for me to compare the difference between the different languages. I don't know ahk very well, so I need to try more!

**Reply by ccz0214** [reply]
大佬 有在做漢化嗎

**Reply by vagrant-soul** [reply]
> 大佬 有在做漢化嗎

正在尝试，最近有点忙，就暂停了几天，等我有时间我会在继续的

**Comment by vagrant-soul** [comment]
Speaking of the path, I'm having this problem too[#494](https://github.com/Lailloken/Lailloken-UI/issues/494), my game is installed in d:\pathofexile
Then I had to change it to d:\pathofexile 2 
^_^

**Reply by xjxhdgb** [reply]
大佬，请问汉化做完了吗，在哪里可以找到啊？

**Comment by Lailloken** [comment]
I released v1.56.4 yesterday evening my time, and it contains a few changes for language detection:
1. the language-setting is now taken from the ini-file rather than the client.txt log-file
2. the language-folder (in your case you picked "chinese") now has to be the same as the code in the ini-file (your previous screenshot shows "zh-TW")

So when the tool starts, it checks the ini-file and gets the code, then checks if there's a folder with the same code (except for English)
- if there is no such folder, the settings menu will show "detected language: not supported"
- if there is such a folder, the settings menu will show "detected language: CODE" and access the correct client.txt language-file (not log-file)

Since it previously couldn't detect the language for PoE2, it always loaded the English client.txt language-file, so maybe that's why you were having issues.
***
If you're seriously considering making a language pack, please open a new thread [here](https://github.com/Lailloken/Lailloken-UI/discussions/categories/translations-localization) and post questions there. This thread is getting a bit long.
